packages/docusaurus/src/client/PendingNavigation.tsx

```diff
@@ -63,12 +63,12 @@ class PendingNavigation extends React.Component<Props, State> {
       // Load data while the old screen remains.
       preload(routes, nextLocation.pathname)
         .then(() => {
-          clientLifecyclesDispatcher.onRouteUpdate({
-            previousLocation: this.previousLocation,
-            location: nextLocation,
-          });
-          // Route has loaded, we can reset previousLocation.
-          this.previousLocation = null;
+          if (this.previousLocation?.pathname !== nextLocation.pathname) {
```

Comment:
not 100% sure about this comparison function (as commented in other issues), but we can give it a try for now

maybe we should factorize this, as it's in 2 places and could diverge